SneakPeek/src/com/cyscorpions/dalejulian/sneakpeek/activities/SneakerEntryListActivity.java

```diff
@@ -0,0 +1,176 @@
+	public static final String LIST_BRAND_EXTRA = "brand";
+	public static final String LIST_SELLING_EXTRA = "selling";
+	public static final String LIST_RARITY_EXTRA = "rarity";
+	public static final String LIST_CATEGORY_EXTRA = "extra";
+	public static final String LIST_DESC_EXTRA = "desc";
+	public static final String LIST_IMGID_EXTRA = "img";
+	public static final String LIST_TITLE_EXTRA = "title";
+	public static final String LIST_ID_EXTRA = "id";
+	public static final String LIST_CATEGORY_EDIT_EXTRA = "editextra";
+	public static final String LIST_FROMLIST_EXTRA = "fromlist";
+
+	private SneakerCategory mCategory = new SneakerCategory();
+
+	@Override
+	protected void onCreate(Bundle savedInstanceState) {
+		super.onCreate(savedInstanceState);
+		setContentView(R.layout.sneaker_entry_listview);
+
+		mCategory.setName(getIntent().getStringExtra(
+				MainActivity.CATEGORY_EXTRA));
+		setTitle("Sneak Peek: " + mCategory.getName());
+
+		mSneakers = SneakerDirectory.get(getApplicationContext())
+				.getSneakersByCategory(mCategory.getName());
+
+		sortEntriesAlphabetically(mSneakers);
+		mAdapter = new SneakerAdapter(this, R.layout.sneaker_entry_listitem,
+				mSneakers);
+		mEntryList = (ListView) findViewById(R.id.listSneakerEntries);
+		mEntryList.setAdapter(mAdapter);
+		mEntryList.setOnItemClickListener(this);
+
+		registerForContextMenu(mEntryList);
+		getActionBar().setDisplayHomeAsUpEnabled(true);
+	}
+
+	@Override
+	public void onItemClick(AdapterView<?> parent, View view, int position,
+			long id) {
+		Sneaker selectedSneaker = mSneakers.get(position);
+		Intent i = new Intent(SneakerEntryListActivity.this,
+				SneakerDetailsActivity.class);
+		i.putExtra(LIST_NAME_EXTRA, selectedSneaker.getName());
+		i.putExtra(LIST_BRAND_EXTRA, selectedSneaker.getBrand());
+		i.putExtra(LIST_SELLING_EXTRA, selectedSneaker.getSellingValue());
+		i.putExtra(LIST_RARITY_EXTRA, selectedSneaker.getRarity());
+		i.putExtra(LIST_CATEGORY_EXTRA, selectedSneaker.getCategory().getName());
+		i.putExtra(LIST_DESC_EXTRA, selectedSneaker.getDescription());
+		i.putExtra(LIST_IMGID_EXTRA, selectedSneaker.getThumbnailId());
+		i.putExtra(LIST_TITLE_EXTRA, selectedSneaker.getTitleName());
+		i.putExtra(LIST_ID_EXTRA, selectedSneaker.getId().toString());
+		startActivityForResult(i, 1);
+
+	}
+
+	@Override
+	public boolean onCreateOptionsMenu(Menu menu) {
+		menu.add("Add Sneak Peek Entry");
+		return super.onCreateOptionsMenu(menu);
+	}
+
+	@Override
+	public boolean onOptionsItemSelected(MenuItem item) {
+		if (item.getTitle() == "Add Sneak Peek Entry") {
+			Intent i = new Intent(SneakerEntryListActivity.this,
+					EditSneakerEntryActivity.class);
+			i.putExtra(LIST_CATEGORY_EDIT_EXTRA, this.mCategory.getName()
+					.toString());
+			startActivityForResult(i, 1);
+		}
+		return super.onOptionsItemSelected(item);
+	}
+
+	@Override
+	protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+		updateList();
```

Comment:
Use a switch statement with each requestCode as a case condition. You can then decide what to do whenever each requestCode returns. In each case condition have an if-else condition for the resultCode which can either be RESULT_OK or RESULT_CANCELLED or you can implement your own resultCode, then decide what will happen in each condition.